workflow/controller/dag.go

```diff
@@ -314,6 +314,14 @@ func (woc *wfOperationCtx) executeDAGTask(dagCtx *dagContext, taskName string) {
 	node := dagCtx.getTaskNode(taskName)
 	task := dagCtx.GetTask(taskName)
 	if node != nil && node.Fulfilled() {
+		// Collect the completed task metrics
+		tmpl := woc.wf.GetTemplateByName(task.Template)
+		if tmpl != nil && tmpl.Metrics != nil {
```

Comment:
Shouldn't this be under the `if node.Completed() {` statement below?

Reply:
it should cover the `Skipped` and `Omitted` status too.